prowler/providers/azure/lib/service/service.py

```diff
@@ -5,13 +5,16 @@
 class AzureService:
     def __init__(
         self,
-        service: str,
+        services: list,
```

Comment:
Could we set here the following? To let callers know that services could be a list or a string.
```suggestion
        services: Union[list, str],
```

Reply:
Also it'd be great to add a comment to the init function to give more details about that.
Like:
```
def __init__():
"""
details line 1
details line 2
"""
```

Reply:
It should not be a string, the most common type of variable is something like this <class 'azure.mgmt.containerservice._container_service_client.ContainerServiceClient'>

Reply:
Good point, in that case it will be better to set something like `Union[list[object], object]` and explain that in the comment.

Reply:
btw `from typing import Union`